sequencer.h

```diff
@@ -108,7 +113,8 @@ struct todo_item {
 	unsigned int flags;
 	int arg_len;
 	/* The offset of the command and its argument in the strbuf */
-	size_t offset_in_buf, arg_offset;
+	size_t offset_in_buf;
+	const char *arg_offset;
```

Comment:
That's not possible because `arg_offset` points into a `strbuf` that can be reallocated at any time, which would render any pointer stale.

In any case, a change like this needs to be refactored into its own commit.

Reply:
I would strongly suggest to stick with offsets.

Reply:
I cannot get a string reference from a `todo_item` if there are only offsets.
Maybe you mean I should keep the `arg_offset` as is, and introduce another field ?

Reply:
I changed the type of `arg_offset` because in the function
```
static int do_pick_commit(struct repository *r,
			  struct todo_item *item,
			  struct replay_opts *opts,
			  int final_fixup, int *check_todo)
```
There is no reference to the current `todo_list` (required to get a reference to the `strbuf` or to call `todo_item_get_arg`).

Am I missing an obvious way to reach a pointer to the current todo_list's `strbuf` ?

Reply:
Just add the `todo_list` as parameter ;-)